lib/src/animated_icons/yaru_animated_icon.dart

```diff
@@ -0,0 +1,107 @@
+import 'package:flutter/material.dart';
+
+enum YaruAnimationMode { once, repeat }
+
+class YaruAnimatedIcon extends StatefulWidget {
+  const YaruAnimatedIcon({
+    required this.data,
+    this.duration,
```

Comment:
Maybe can we make this parameter a non-named one?
Like `Text` and `Icon` widgets?

```suggestion
  const YaruAnimatedIcon(
    this.data, {
    this.duration,
```

So we can write:

```dart
YaruAnimatedIcon(YaruAnimatedOkIcon())
```

As this will be the most common use.